appendices/symbol-lookup-cheatsheet.md

```diff
@@ -17,3 +17,55 @@ what you should search the tutorial for in order to learn more about the symbol.
 | ?  | Partial function |
 | '  | Prime |
 | <: | Subtype |
+
+
+Here is a more elaborate explanation of Pony's use of special characters:
+
+|Symbol | Usage|
+| --- | --- |
+| ,  | to separate parameters in a function signature, or the items of a tuple
+| .  | (1) to call a field or a function on a variable (field access or method call)
+     | (2) to qualify a type/method with its package name
+| '  | used as alternative name in parameters (prime)
+| "  | to delineate a literal string
+| """  | to delineate a documentation string
+| (  | (1) start of line: start of a tuple
+     | (2) middle of line: method call
+| () | (1) parentheses, for function or behavior parameters 
+     | (2) making a tuple (values separated by ,)
+     | (3) making an enumeration (values separated by |)
+| [  | (1) start of line: start of an array literal
+     | (2) middle of line: generic formal parameters
+| [ ]  | (1) to indicate a generic type, for example Range[U64]
+       | (2) to indicate the return type of an FFI function call
+| { }  | a function type
+| :  | (1) after a variable: is followed by the type name 
+     | (2) to indicate a function return type
+|    | (3) a type constraint
```

Comment:
Indeed, this is an error; I will correct it.

 

From: Sean T Allen [mailto:notifications@github.com] 
Sent: maandag 11 april 2016 14:29
To: ponylang/pony-tutorial <pony-tutorial@noreply.github.com>
Cc: ibalbaert <ivo.balbaert@telenet.be>
Subject: Re: [ponylang/pony-tutorial] Added 2nd usage section for symbol cheat-sheet (#91)

 

In appendices/symbol-lookup-cheatsheet.md <https://github.com/ponylang/pony-tutorial/pull/91#discussion_r59195342> :

> +| '  | used as alternative name in parameters (prime)
> +| "  | to delineate a literal string
> +| """  | to delineate a documentation string
> +| (  | (1) start of line: start of a tuple
> +     | (2) middle of line: method call
> +| () | (1) parentheses, for function or behavior parameters 
> +     | (2) making a tuple (values separated by ,)
> +     | (3) making an enumeration (values separated by |)
> +| [  | (1) start of line: start of an array literal
> +     | (2) middle of line: generic formal parameters
> +| [ ]  | (1) to indicate a generic type, for example Range[U64]
> +       | (2) to indicate the return type of an FFI function call
> +| { }  | a function type
> +| :  | (1) after a variable: is followed by the type name 
> +     | (2) to indicate a function return type
> +|    | (3) a type constraint

Has | but no symbol. Is this to indicate its a ':' ?

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub <https://github.com/ponylang/pony-tutorial/pull/91/files/80196091b4e62a5b9b923bb290ab4aac68909f35#r59195342>   <https://github.com/notifications/beacon/AA6zfPR-Gby_-WIHFBbsEeGlXfW6tAqMks5p2j57gaJpZM4IEQOw.gif> 



---
Dit e-mailbericht is gecontroleerd op virussen met Avast antivirussoftware.
https://www.avast.com/antivirus